classes/migration/upgrade/v3_4_0/I8933_EventLogLocalized.php

```diff
@@ -64,41 +66,56 @@ public function up(): void
         $this->renameSettings();
 
         // Localize existing submission file name entries
-        $sitePrimaryLocale = DB::table('site')->value('primary_locale');
+        $sitePrimaryLocale = DB::getPdo()->quote(DB::table('site')->value('primary_locale'));
         $contexts = DB::table($this->getContextTable())->get([$this->getContextIdColumn(), 'primary_locale']);
-        $contextIdPrimaryLocaleMap = [];
-        foreach ($contexts as $context) {
-            $contextIdPrimaryLocaleMap[$context->{$this->getContextIdColumn()}] = $context->primary_locale;
-        }
+        $whenRules = $contexts
+            ->map(fn (object $context) => 'WHEN ' . (int) $context->{$this->getContextIdColumn()} . ' THEN ' . DB::getPdo()
+            ->quote($context->primary_locale ?? $sitePrimaryLocale))->join("\n");
+
+        // Attempts to retrieve a context ID from the event_log.assoc_id
+        $contextIdQuery = DB::table('submissions AS s')
```

Comment:
The `$contextIdQuery` could be a `leftJoin()` on the next query, but the `updateFrom()` turns it into a `inner join` (sounds like a Laravel bug).